### ISSUE TITLE
Injectable renderInput for MobileKeyboardInputView

### ISSUE THREAD

**Issue by ph-fritsche** [mui-org/material-ui-pickers#1932]
**Is your feature request related to a problem? Please describe.**
When using the picker with an own `renderInput`, that `renderInput` is also used for rendering `MobileKeyboardInputView`, while the component style and controls might not be fitting inside the `PickerModalDialog`.

**Describe the solution you'd like**
Make either the InputComponent for `MobileKeyboardInputView` or the render function used by it injectable.

```jsx
<Picker
  renderInput={myDefaultRender}
  renderModalInput={mySimpleRender}
/>
```

**Describe alternatives you've considered**
With `4.0.0-alpha.8` you can inject `PureDateInputComponent` and `KeyboardDateInputComponent`. (Not documented.)
Then you can replace `renderInput` with the render function for the [`MobileKeyboardInputView` because it always uses the built in `KeyboardDateInput`](https://github.com/mui-org/material-ui-pickers/blob/21ef115f26236f3367fc39294d2063ebf580a352/lib/src/Picker/Picker.tsx#L159).

```jsx
      <Picker
        renderInput={renderSimple}
        PureDateInputComponent={renderSpecial}
        KeyboardDateInputComponent={renderSpecial}
      />
```
https://codesandbox.io/s/pickerinputcomponent-5rqyo?file=/src/App.js

But this solution is very hacky as it replaces exactly what you want to keep (and you can not reuse the internal `PureDateInput` and `KeyboardDateInput` as they are not exported).

**Comment by dmtrKovalenko** [comment]
What the underlying problem your trying to solve? Why do you need to style mobile keyboard input differently?

**Comment by ph-fritsche** [comment]
I've got a solution for a `TextField` with a customized keyboard-friendly inputComponent that is look- and feel-wise consistent with other fields (file upload, text, numbers, etc.) around.
I'd like to use the `Picker`s for the non-keyboard-users on desktop and mobile.

If a user switches to `MobileKeyboardInputView` he sees the same component as in the root document, but I think that component is not the best representation inside the `Modal`.

The solution I turned to is implementing the `Modal` myself, rendering the `TextField` outside of the `Static` `Picker` and only passing the `renderInput` for the `MobileKeyboardInputView`, but I'd rather reuse the whole `Picker` component.
